alvr/server/src/connection.rs

```diff
@@ -556,37 +556,35 @@ fn connection_pipeline(
-            )
-            .to_con()?;
-
             #[cfg(not(target_os = "linux"))]
-            if let Switch::Enabled(microphone_config) = &settings.audio.microphone {
-                let (sink, source) = AudioDevice::new_virtual_microphone_pair(
-                    Some(settings.audio.linux_backend),
-                    microphone_config.devices.clone(),
-                )
-                .to_con()?;
-                if matches!(
-                    microphone_config.devices,
-                    alvr_session::MicrophoneDevicesConfig::VBCable
-                ) {
-                    // VoiceMeeter and Custom devices may have arbitrary internal routing.
-                    // Therefore, we cannot detect the loopback issue without knowing the routing.
-                    if alvr_audio::is_same_device(&game_audio_device, &sink)
-                        || alvr_audio::is_same_device(&game_audio_device, &source)
-                    {
-                        con_bail!("Game audio and microphone cannot point to the same device!");
+            {
+                let game_audio_device =
+                    AudioDevice::new_output(game_audio_config.device.as_ref()).to_con()?;
+                if let Switch::Enabled(microphone_config) = &settings.audio.microphone {
+                    let (sink, source) =
+                        AudioDevice::new_virtual_microphone_pair(microphone_config.devices.clone())
+                            .to_con()?;
+                    if matches!(
+                        microphone_config.devices,
+                        alvr_session::MicrophoneDevicesConfig::VBCable
+                    ) {
+                        // VoiceMeeter and Custom devices may have arbitrary internal routing.
+                        // Therefore, we cannot detect the loopback issue without knowing the routing.
+                        if alvr_audio::is_same_device(&game_audio_device, &sink)
+                            || alvr_audio::is_same_device(&game_audio_device, &source)
+                        {
+                            con_bail!("Game audio and microphone cannot point to the same device!");
+                        }
                     }
+                    // else:
+                    // Stream played via VA-CABLE-X will be directly routed to VA-CABLE-X's virtual microphone.
+                    // Game audio will loop back to the game microphone if they are set to the same VA-CABLE-X device.
                 }
-                // else:
-                // Stream played via VA-CABLE-X will be directly routed to VA-CABLE-X's virtual microphone.
-                // Game audio will loop back to the game microphone if they are set to the same VA-CABLE-X device.
-            }
 
-            game_audio_device.input_sample_rate().to_con()?
+                game_audio_device.input_sample_rate().to_con()?
+            }
+            #[cfg(target_os = "linux")]
+            44100
```

Comment:
Forgot to remove this hardcoded value

Reply:
it's not forgotten, it's only place where it gets sample rate for pipewire

Either we hardcode or we somehow need to ask pipewire for it (not sure how for now)

Reply:
Alright